### PR TITLE
Fix tests

### PR DESCRIPTION
**Pull request by slabgorb**
Refactor, split into smaller helper functions to show work better. Make 'only-works-if-it-spans-one-other-day' into a recursive function. Add many tests and make them more targeted to function rather than all at once. 

Some restrictions imposed to parameters and constructors. Instead of params being passed into the functions, made them methods of the class.

### CODE REVIEW

File: test_working_days.py [hunk 1]
```diff
@@ -0,0 +1,41 @@
+import pendulum
+from sla_calculator import SLA_Calculator
+
+holiday_date = pendulum.datetime(2022, 1, 3, tz=pendulum.local_timezone())
```
Comment (slabgorb) [inline test_working_days.py]: I guess these are technically constants *sigh* so I should allcaps them

Comment (slabgorb) [reply test_working_days.py]: it's teeeeeeeessssssstttttssss though

File: sla_calculator/calculator.py [hunk 2]
```diff
@@ -2,62 +2,172 @@
...
+                0,
+                tz=pendulum.local_timezone(),
+            )
+        time_left_today = self._calculate_time_left_today(start_time)
+        if time_left_today >= sla:
+            sla_time = start_time.add(minutes=sla)
+        else:
+            next_day = self._next_day_open_time(start_time)
+            sla = sla - time_left_today
+            return self.calculate(next_day, minutes=sla)
```
Comment (slabgorb) [inline sla_calculator/calculator.py]: recursion in business logic!!!!

**Review by Zious11 — APPROVED** [review]

**Pull request merged**